ec/src/hashing/curve_maps/swu.rs

```diff
@@ -99,7 +86,7 @@ impl<P: SWUConfig> MapToCurve<Projective<P>> for SWUMap<P> {
         let gx1_square;
         let gx1;
 
-        assert!(
+        debug_assert!(
```

Comment:
What was the motivation of changing this to `debug_assert`?

Reply:
This one is a little more tricky, @drskalman maybe can confirm, but it should never happen at this point that the assert fails right?

Reply:
So, technically under the current code, it could happen that `div = 0`. [Here](https://github.com/arkworks-rs/algebra/pull/659/files#diff-1334780631de1d3814e5a555c6064b485165e7e7b63620c58214ff9209894b7dR73) we apparently get a non-zero `div`, but that's working off the assumption that `P::ZETA` is non-zero.
The current check  on `ZETA` [here](https://github.com/arkworks-rs/algebra/blob/0c945fa0210d70c17f93a42af3e0a25e7e5b757f/ec/src/hashing/curve_maps/swu.rs#L31) should actually be more strict - currently it allows a zero, whereas according to [RFC9380 Sec. 6.6.2](https://www.rfc-editor.org/rfc/rfc9380.pdf): "Z is non-square in F".
With the stricter condition, this `debug_assert` here is fine instead of `assert`.

Reply:
I guess my higher-level question was that earlier these checks would happen even in release mode; is it safe for them to happen only in debug mode now?

Reply:
So it should not fail if the map is correct. So, it is safe if the map is correct. However if an implementor choose a wrong constant, I thought it is a good idea that they get the error that they are mapping off the curve.

However, once they got the paramaters right, then we could forgo the check in the interest of performance assuming that they have at least run it once in the debug mode.
Given that it never fails if the parameters are right, we could remove the check completely if you don't like `debug_assert` rather than slowing done every hashing with `assert`. But I thought we can give a hand to the curve developers in debug time so they can fix their maps.

Your call.

Reply:
Makes sense to me, thanks for the explanation!